ts/resolution.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2014 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Matt Tracy (matt.r.tracy@gmail.com)
+
+package ts
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/proto"
+)
+
+// Resolution is used to enumerate the different resolution values supported by
+// cockroach.
+type Resolution int64
+
+const (
+	// Resolution10s stores data with a sample resolution of 10 seconds.
+	Resolution10s Resolution = iota
```

Comment:
Now that you are encoding the `Resolution` value into the key, I think you should avoid using `iota` and manually specify the enum values. Also add a comment that the values must not change since they are stored in the encoded keys.